Tools/Scripts/libraries/reporelaypy/reporelaypy/checkout.py

```diff
@@ -249,32 +249,36 @@ def push_update(self, branch=None, tag=None, remote=None, track=False, dest_bran
         num_commits = 1 if sentinel.returncode else int(sentinel.stdout.rstrip())
         while num_commits > 1 and (num_commits < 50 or branch == self.repository.default_branch):
             num_commits -= 1
-            if run(
+            run(
                 [self.repository.executable(), 'push', remote, '{}~{}:{}'.format(branch, num_commits, dest_branch or branch)],
                 cwd=self.repository.root_path,
-            ).returncode:
-                break
+            )
 
         return not run(
             [self.repository.executable(), 'push', remote, '{}:{}'.format(branch, dest_branch or branch)],
             cwd=self.repository.root_path,
         ).returncode
 
     def forward_update(self, branch=None, tag=None, remote=None, track=False):
+        did_succeed_all = True
+
         for match, tos in self.forwarding:
             if match not in (remote, '{}:{}'.format(remote, branch or tag)):
                 continue
             for to in tos:
                 split = to.split(':', 1)
-                self.push_update(
+                did_succeed_this = self.push_update(
                     branch=branch,
                     tag=tag,
                     remote=split[0],
                     track=track,
                     dest_branch=split[1] if len(split) > 1 else None,
                 )
-            return
-        return
+                if not did_succeed_this:
+                    sys.stderr.write(f"Failed to push to '{split[0]}'\n")
+                did_succeed_all = did_succeed_all and did_succeed_this
+            return did_succeed_all
```

Comment:
Is leaving this line in a typo?

Reply:
It's not a typo. This _may_ be a bug, it's not obviously so (the early return means we only consume one forwarding rule), but there is an ongoing problem that points to this line possibly being incorrect. I'm still actively investigating.